spec/index.html

```diff
@@ -1466,164 +1473,233 @@ <h2>Generalizations of RDF Triples, Graphs, and Datasets</h2>
-      It returns a <a>blank node</a> <var>b</var>,
-      a map <var>Mₒ</var> from <a>triple terms</a> to <a>blank nodes</a>,
-      and a [=Classic=] <a>RDF dataset</a> <var>D</var>.
-      In the algorithm, we adopt the view presented in <a href="#section-dataset-quad"></a>.
+  </section>
+
+  <section id="section-unclassicize-definition">
+    <h2>From [=Classic=] to [=Full=]</h2>
+
+    <p>Reverting a [=classicize=] graph to its original form consists,
+      for each [=blank node=] <var>b</var> that is the subject of an [=asserted=] triple (<var>b</var>, `rdf:type`, `rdf:TripleTerm`),
+      in locating the three other [=asserted=] triples (<var>b</var>, `rdf:ttSubject`, <var>s</var>),
+      (<var>b</var>, `rdf:ttPredicate`, <var>p</var>),
+      and (<var>b</var>, `rdf:ttObject`, <var>o</var>).
+      These four triples are removed from the graph.
+      All remaining occurrences of <var>b</var> as a [=constituent term=] of the graph are then replaced with the triple term (<var>s</var>, <var>p</var>, <var>o</var>).
     </p>
 
-    <ol>
-      <li>Let <var>Mₒ</var> be an empty map.</li>
-      <li>Let <var>D</var> be an empty <a>RDF dataset</a>.</li>
-      <li>If <var>Mᵢ</var> contains a <a>blank node</a> <var>b</var> associated with <var>t</var>, then return <var>b</var>, <var>Mₒ</var> and <var>D</var>.
-      <li>Otherwise:<ol>
-        <li>Let <var>s</var>, <var>p</var> and <var>o</var> be the subject, predicate and object of <var>t</var>, respectively.</li>
-        <li>If <var>o</var> is a <a>triple term</a>, then:<ol>
-          <li>Let <var>b'</var>, <var>M'</var> and <var>D'</var> be the result of invoking <a href="#section-qtt-algo">`quote-triple-term`</a> passing <var>o</var> as <var>t</var> and <var>Mᵢ</var>.</li>
-          <li>Set <var>o</var> to <var>b'</var>.
-          <li>Merge <var>M'</var> into <var>Mₒ</var>.
-          <li>Merge <var>D'</var> into <var>D</var>.
-        </ol></li>
-        <li id="qtt-fresh-bnode">Let <var>b</var> be a fresh blank node.</li>
-        <li>Add the association (<var>t</var>, <var>b</var>) to <var>Mₒ</var>.</li>
-        <li>Add the quads (<var>s</var>, <var>p</var>, <var>o</var>, <var>b</var>) and (<var>b</var>, `rdf:type`, `rdf:TripleTerm`, `rdf:unstarMetadata`) in <var>D</var>.</li>
-        <li>Return <var>b</var>, <var>Mₒ</var> and <var>D</var>.</li>
-      </ol></li>
-    </ol>
+    <p>Implementations MUST report an error if, for a given <var>b</var>,
+      it can not unambiguously determine <var>s</var>, <var>p</var> or <var>o</var>
+      (i.e. if one of the `classicize:` properties of <var>b</var> is missing or duplicated).
+      Implementations MUST also report an error if the input graph contains at the same time a [=triple term=] and an [=asserted triple=] (<var>b</var>, `rdf:type`, `rdf:TripleTerm`) where <var>b</var> is a [=blank node=].
+      None of these situations can occur if the input graph was produced by the [=classicize=] transformation.
```

Comment:
```suggestion
    <p>An implementation MUST report an error if, for a given <var>b</var>,
      it can not unambiguously determine <var>s</var>, <var>p</var>, or <var>o</var>
      (i.e., if one of the `classicize:` properties
      — `rdf:ttSubject`, `rdf:ttPredicate`, or `rdf:ttObject` —
      of <var>b</var> is missing or duplicated).
      An implementation MUST also report an error if the input graph contains
      at the same time a [=triple term=] and an [=asserted triple=]
      (<var>b</var>, `rdf:type`, `rdf:TripleTerm`) 
      where <var>b</var> is the same [=blank node=].
      Note that none of these situations can occur if the input graph was produced by the [=classicize=] transformation.
```